### ISSUE TITLE
Access denied when adding url from Netlify to allowed_origins

### ISSUE THREAD

**Issue by seel93** [aws-samples/amazon-cognito-passwordless-auth#46]
We have moved from hosting our vue-app in s3 to Netlify, now the passwordless feature fails with the following error:

{"__type":"UnexpectedLambdaException","message":"VerifyAuthChallengeResponse invocation failed due to error AccessDeniedException."}

There are no changes in the app besides moving the hosting

**Comment by ottokruse** [comment]
Best check the logs of the Verify Auth challenge lambda function? It's that lambda function that reports it does not have access in your message. Hopefully the stack trace in the logs reports what the lambda function was doing. Otherwise set loglevel to Debug.

Looks like it's unrelated to the Netlify move by the way?

**Comment by ottokruse** [comment]
Hard to follow what it happening. Maybe the User Pool does not have permission to invoke the lambda function. This permission is created automatically by the CDK stack but maybe got inadvertantly removed. (Still I see no reason why this would be related to a move from S3 to Netlify)

**Comment by ottokruse** [comment]
Any luck @seel93 ?
If not can you provide more details? Eg show how you use the Passwordless construct and how you deploy. Ideally we can reproduce. Also, what version of the passwordless lib are you on ?

**Comment by seel93** [comment]
We're only using a small part of the code in this repo and using terraform instead of the cdk. 

Magiclink is currently the only sign-in method we're using, and code for all the other methods is not a part of our repo.

This solution was working fine with s3 and cloudfront, and when we introduced Netlify the verifyAuth trigger failed to run.

I don't have the passwordless lib version in front of me right now, but will provide it asap.

**Comment by seel93** [comment]
@ottokruse  we're using v0.5.1 of the passwordless lib

**Comment by ottokruse** [comment]
Check the resource based policy statements of the VerifyAuth lambda in the AWS console and make sure correct?

Should be something like:

<img width="1246" alt="image" src="https://github.com/aws-samples/amazon-cognito-passwordless-auth/assets/6275520/7395d42a-c246-401a-b50c-5e2588ffb1d4">

And when opened:

![image](https://github.com/aws-samples/amazon-cognito-passwordless-auth/assets/6275520/b5cea995-f12a-4216-9a9d-9e717fbfd2ed)

**Comment by seel93** [comment]
@ottokruse This issue was linked to setting the wrong client_id for cognito in netlify. The correct id has been set for the env vars and the solution works as intended.

Apologies for the inconvenience!
